### PR TITLE
buildDotnetModule: inherit enableParallelBuilding

### PR DESCRIPTION
**Pull request by tie**
## Description of changes

After this change, enableParallelBuilding defaults to true for packages using buildDotnetModule. The argument value already defaults to true, but we did not use it. Note that default values in Nix are not present in the arguments attribute set, e.g. `args` does not contain `def` for `{ def ? true }@args: …` function unless this argument is explicitly passed.

__References__:
- https://github.com/NixOS/nixpkgs/pull/339220#issuecomment-2327318933

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tie** [comment]
Result of `nixpkgs-review pr 339335` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>naps2</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>nexusmods-app</li>
    <li>nexusmods-app-unfree</li>
    <li>recyclarr</li>
    <li>ryujinx</li>
  </ul>
</details>
<details>
  <summary>66 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>am2rlauncher</li>
    <li>avalonia-ilspy</li>
    <li>beatsabermodmanager</li>
    <li>bicep</li>
    <li>blendfarm</li>
    <li>boogie</li>
    <li>btcpayserver</li>
    <li>btcpayserver-altcoins</li>
    <li>cavalier</li>
    <li>celeste64</li>
    <li>certdump</li>
    <li>csharp-ls</li>
    <li>csharpier</li>
    <li>csharprepl</li>
    <li>cyclonedx-cli</li>
    <li>dafny</li>
    <li>denaro</li>
    <li>depotdownloader</li>
    <li>dotnet-outdated</li>
    <li>fable</li>
    <li>famistudio</li>
    <li>fantomas</li>
    <li>formula</li>
    <li>fsautocomplete</li>
    <li>galaxy-buds-client</li>
    <li>garnet</li>
    <li>gh-gei</li>
    <li>git-credential-manager</li>
    <li>github-runner</li>
    <li>gitversion</li>
    <li>ilspycmd</li>
    <li>jackett</li>
    <li>jellyfin</li>
    <li>juniper</li>
    <li>kavita</li>
    <li>knossosnet</li>
    <li>kryptor</li>
    <li>libation</li>
    <li>lubelogger</li>
    <li>lumafly</li>
    <li>marksman</li>
    <li>mqttmultimeter</li>
    <li>msbuild</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>networkminer</li>
    <li>omnisharp-roslyn</li>
    <li>opentabletdriver</li>
    <li>openutau</li>
    <li>pablodraw</li>
    <li>parabolic</li>
    <li>pbm</li>
    <li>pinta</li>
    <li>pupdate</li>
    <li>retrospy</li>
    <li>roslyn</li>
    <li>roslyn-ls</li>
    <li>scarab</li>
    <li>slskd</li>
    <li>sonarr</li>
    <li>tagger</li>
    <li>technitium-dns-server</li>
    <li>tone</li>
    <li>torrentstream</li>
    <li>wiseunpacker</li>
  </ul>
</details>

**Comment by tie** [comment]
Result of `nixpkgs-review pr 339335` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>naps2</li>
  </ul>
</details>
<details>
  <summary>70 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>am2rlauncher</li>
    <li>avalonia-ilspy</li>
    <li>beatsabermodmanager</li>
    <li>bicep</li>
    <li>blendfarm</li>
    <li>boogie</li>
    <li>btcpayserver</li>
    <li>btcpayserver-altcoins</li>
    <li>cavalier</li>
    <li>celeste64</li>
    <li>certdump</li>
    <li>csharp-ls</li>
    <li>csharpier</li>
    <li>csharprepl</li>
    <li>cyclonedx-cli</li>
    <li>dafny</li>
    <li>denaro</li>
    <li>depotdownloader</li>
    <li>dotnet-outdated</li>
    <li>fable</li>
    <li>famistudio</li>
    <li>fantomas</li>
    <li>formula</li>
    <li>fsautocomplete</li>
    <li>galaxy-buds-client</li>
    <li>garnet</li>
    <li>gh-gei</li>
    <li>git-credential-manager</li>
    <li>github-runner</li>
    <li>gitversion</li>
    <li>ilspycmd</li>
    <li>jackett</li>
    <li>jellyfin</li>
    <li>juniper</li>
    <li>kavita</li>
    <li>knossosnet</li>
    <li>kryptor</li>
    <li>libation</li>
    <li>lubelogger</li>
    <li>lumafly</li>
    <li>marksman</li>
    <li>mqttmultimeter</li>
    <li>msbuild</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>networkminer</li>
    <li>nexusmods-app</li>
    <li>nexusmods-app-unfree</li>
    <li>omnisharp-roslyn</li>
    <li>opentabletdriver</li>
    <li>openutau</li>
    <li>pablodraw</li>
    <li>parabolic</li>
    <li>pbm</li>
    <li>pinta</li>
    <li>pupdate</li>
    <li>recyclarr</li>
    <li>retrospy</li>
    <li>roslyn</li>
    <li>roslyn-ls</li>
    <li>ryujinx</li>
    <li>scarab</li>
    <li>slskd</li>
    <li>sonarr</li>
    <li>tagger</li>
    <li>technitium-dns-server</li>
    <li>tone</li>
    <li>torrentstream</li>
    <li>wiseunpacker</li>
  </ul>
</details>

**Review by corngood — APPROVED** [review]
Good find.  Most of `@args` is a bit of a mess.  There are a lot of unused default values, as well as things that should be stripped before stdenv.  At least it looks like the rest of the default values are all null/false/[].

Cc: @NixOS/dotnet

File: pkgs/by-name/ne/nexusmods-app/package.nix [hunk 1]
```diff
@@ -25,6 +25,8 @@ buildDotnetModule (finalAttrs: {
     hash = "sha256-FzQphMhiC1g+6qmk/R1v4rq2ldy35NcaWm0RR1UlwLA=";
   };
 
+  enableParallelBuilding = false;
```
Comment (MattSturgeon) [inline pkgs/by-name/ne/nexusmods-app/package.nix:28]: I assume you tried to build in parallel but it failed?

**Review by MattSturgeon — APPROVED** [review]

**Comment by corngood** [comment]
Result of `nixpkgs-review pr 339335` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>certdump</li>
    <li>pablodraw</li>
  </ul>
</details>
<details>
  <summary>45 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>avalonia-ilspy</li>
    <li>azure-functions-core-tools</li>
    <li>bicep</li>
    <li>blendfarm</li>
    <li>boogie (dotnetPackages.Boogie)</li>
    <li>btcpayserver</li>
    <li>btcpayserver-altcoins</li>
    <li>csharp-ls</li>
    <li>csharpier</li>
    <li>csharprepl</li>
    <li>cyclonedx-cli</li>
    <li>dafny</li>
    <li>depotdownloader</li>
    <li>dotnet-outdated</li>
    <li>famistudio</li>
    <li>fantomas</li>
    <li>formula</li>
    <li>fsautocomplete</li>
    <li>garnet</li>
    <li>gh-gei</li>
    <li>git-credential-manager</li>
    <li>github-runner</li>
    <li>gitversion</li>
    <li>ilspycmd</li>
    <li>jackett</li>
    <li>jellyfin</li>
    <li>juniper</li>
    <li>knossosnet</li>
    <li>kryptor</li>
    <li>libation</li>
    <li>lubelogger</li>
    <li>marksman</li>
    <li>msbuild</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>omnisharp-roslyn</li>
    <li>openutau</li>
    <li>recyclarr</li>
    <li>roslyn</li>
    <li>roslyn-ls</li>
    <li>sonarr</li>
    <li>tests.dotnet.structured-attrs.no-structured-attrs</li>
    <li>torrentstream</li>
    <li>wiseunpacker</li>
  </ul>
</details>

File: pkgs/by-name/ne/nexusmods-app/package.nix (hunk 1 above)
Comment (corngood) [reply pkgs/by-name/ne/nexusmods-app/package.nix:28]: It was in the first nixpkgs-review, on aarch64-linux.  Something we should look into.

**Comment by corngood** [comment]
Result of `nixpkgs-review pr 339335` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>93 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>alttpr-opentracker</li>
    <li>am2rlauncher</li>
    <li>avalonia-ilspy</li>
    <li>azure-functions-core-tools</li>
    <li>beatsabermodmanager</li>
    <li>bicep</li>
    <li>blendfarm</li>
    <li>boogie (dotnetPackages.Boogie)</li>
    <li>btcpayserver</li>
    <li>btcpayserver-altcoins</li>
    <li>cavalier</li>
    <li>celeste64</li>
    <li>certdump</li>
    <li>csharp-ls</li>
    <li>csharpier</li>
    <li>csharprepl</li>
    <li>cyclonedx-cli</li>
    <li>dafny</li>
    <li>denaro</li>
    <li>depotdownloader</li>
    <li>discordchatexporter-cli</li>
    <li>dotnet-outdated</li>
    <li>eventstore</li>
    <li>fable</li>
    <li>famistudio</li>
    <li>fantomas</li>
    <li>formula</li>
    <li>fsautocomplete</li>
    <li>galaxy-buds-client</li>
    <li>garnet</li>
    <li>gh-gei</li>
    <li>git-credential-manager</li>
    <li>github-runner</li>
    <li>gitversion</li>
    <li>ilspycmd</li>
    <li>inklecate</li>
    <li>jackett</li>
    <li>jellyfin</li>
    <li>juniper</li>
    <li>kavita</li>
    <li>knossosnet</li>
    <li>kryptor</li>
    <li>libation</li>
    <li>lubelogger</li>
    <li>lumafly</li>
    <li>marksman</li>
    <li>mqttmultimeter</li>
    <li>msbuild</li>
    <li>naps2</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>networkminer</li>
    <li>nexusmods-app</li>
    <li>nexusmods-app-unfree</li>
    <li>omnisharp-roslyn</li>
    <li>openra</li>
    <li>opentabletdriver</li>
    <li>openutau</li>
    <li>osu-lazer</li>
    <li>pablodraw</li>
    <li>parabolic</li>
    <li>pbm</li>
    <li>pinta</li>
    <li>ps3-disc-dumper</li>
    <li>pupdate</li>
    <li>recyclarr</li>
    <li>retrospy</li>
    <li>roslyn</li>
    <li>roslyn-ls</li>
    <li>ryujinx</li>
    <li>scarab</li>
    <li>slskd</li>
    <li>sonarr</li>
    <li>space-station-14-launcher</li>
    <li>tagger</li>
    <li>technitium-dns-server</li>
    <li>tests.dotnet.final-attrs.check-output</li>
    <li>tests.dotnet.final-attrs.output-matches-const</li>
    <li>tests.dotnet.final-attrs.override-has-no-effect</li>
    <li>tests.dotnet.final-attrs.override-modifies-output</li>
    <li>tests.dotnet.project-references</li>
    <li>tests.dotnet.structured-attrs.check-output</li>
    <li>tests.dotnet.structured-attrs.no-structured-attrs</li>
    <li>tests.dotnet.use-dotnet-from-env.fallback</li>
    <li>tests.dotnet.use-dotnet-from-env.use-dotnet-path-env</li>
    <li>tests.dotnet.use-dotnet-from-env.use-dotnet-root-env</li>
    <li>tests.dotnet.use-dotnet-from-env.without-fallback</li>
    <li>tone</li>
    <li>torrentstream</li>
    <li>wasabibackend</li>
    <li>wiseunpacker</li>
    <li>xivlauncher</li>
  </ul>
</details>

**Comment by corngood** [comment]
Result of `nixpkgs-review pr 339335` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>93 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>alttpr-opentracker</li>
    <li>am2rlauncher</li>
    <li>avalonia-ilspy</li>
    <li>azure-functions-core-tools</li>
    <li>beatsabermodmanager</li>
    <li>bicep</li>
    <li>blendfarm</li>
    <li>boogie (dotnetPackages.Boogie)</li>
    <li>btcpayserver</li>
    <li>btcpayserver-altcoins</li>
    <li>cavalier</li>
    <li>celeste64</li>
    <li>certdump</li>
    <li>csharp-ls</li>
    <li>csharpier</li>
    <li>csharprepl</li>
    <li>cyclonedx-cli</li>
    <li>dafny</li>
    <li>denaro</li>
    <li>depotdownloader</li>
    <li>discordchatexporter-cli</li>
    <li>dotnet-outdated</li>
    <li>eventstore</li>
    <li>fable</li>
    <li>famistudio</li>
    <li>fantomas</li>
    <li>formula</li>
    <li>fsautocomplete</li>
    <li>galaxy-buds-client</li>
    <li>garnet</li>
    <li>gh-gei</li>
    <li>git-credential-manager</li>
    <li>github-runner</li>
    <li>gitversion</li>
    <li>ilspycmd</li>
    <li>inklecate</li>
    <li>jackett</li>
    <li>jellyfin</li>
    <li>juniper</li>
    <li>kavita</li>
    <li>knossosnet</li>
    <li>kryptor</li>
    <li>libation</li>
    <li>lubelogger</li>
    <li>lumafly</li>
    <li>marksman</li>
    <li>mqttmultimeter</li>
    <li>msbuild</li>
    <li>naps2</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>networkminer</li>
    <li>nexusmods-app</li>
    <li>nexusmods-app-unfree</li>
    <li>omnisharp-roslyn</li>
    <li>openra</li>
    <li>opentabletdriver</li>
    <li>openutau</li>
    <li>osu-lazer</li>
    <li>pablodraw</li>
    <li>parabolic</li>
    <li>pbm</li>
    <li>pinta</li>
    <li>ps3-disc-dumper</li>
    <li>pupdate</li>
    <li>recyclarr</li>
    <li>retrospy</li>
    <li>roslyn</li>
    <li>roslyn-ls</li>
    <li>ryujinx</li>
    <li>scarab</li>
    <li>slskd</li>
    <li>sonarr</li>
    <li>space-station-14-launcher</li>
    <li>tagger</li>
    <li>technitium-dns-server</li>
    <li>tests.dotnet.final-attrs.check-output</li>
    <li>tests.dotnet.final-attrs.output-matches-const</li>
    <li>tests.dotnet.final-attrs.override-has-no-effect</li>
    <li>tests.dotnet.final-attrs.override-modifies-output</li>
    <li>tests.dotnet.project-references</li>
    <li>tests.dotnet.structured-attrs.check-output</li>
    <li>tests.dotnet.structured-attrs.no-structured-attrs</li>
    <li>tests.dotnet.use-dotnet-from-env.fallback</li>
    <li>tests.dotnet.use-dotnet-from-env.use-dotnet-path-env</li>
    <li>tests.dotnet.use-dotnet-from-env.use-dotnet-root-env</li>
    <li>tests.dotnet.use-dotnet-from-env.without-fallback</li>
    <li>tone</li>
    <li>torrentstream</li>
    <li>wasabibackend</li>
    <li>wiseunpacker</li>
    <li>xivlauncher</li>
  </ul>
</details>

**Pull request merged**